BookAngular/src/app/services/book-service.service.ts

```diff
@@ -1,71 +1,71 @@
-import { Injectable } from '@angular/core';
-import { HttpClient } from "@angular/common/http";
-import { BookModel } from '../model/BookModel';
-import { Book } from "../model/Book";
-import { Observable, Observer } from 'rxjs';
-@Injectable({
-  providedIn: 'root'
-})
+  import { Injectable } from '@angular/core';
+  import { HttpClient } from "@angular/common/http";
+  import { BookModel } from '../model/BookModel';
+  import { Book } from "../model/Book";
+  import { Observable, Observer } from 'rxjs';
+  @Injectable({
+    providedIn: 'root'
+  })
 
-export class BookServiceService {
+  export class BookServiceService {
 
-  constructor(private http: HttpClient) { }
+    constructor(private http: HttpClient) { }
 
-  listBooks: BookModel[];
-  baseUrl: string = "https://localhost:44317/api/ProjectCompare/book";
-  baseUrlPaging: string = "https://localhost:44317/api/ProjectCompare/book?pageSize=5&pageNumber=1";
-  
-  GetBook(pageSize?: number, pageNumber?: number, searchString? : string): Observable<BookModel> {
-    let result: any;
-    //Non Search
-    if(searchString == null){
-      if (pageSize !== undefined && pageNumber !== undefined) {
-        const urlPaging = `${this.baseUrl}/?pageSize=${pageSize}&pageNumber=${pageNumber}`
-        result = this.http.get(urlPaging);
-        return result;
-      } else {
-        result = this.http.get(this.baseUrlPaging);
-        return result
+    listBooks: BookModel[];
+    baseUrl: string = "https://localhost:44317/api/ProjectCompare/book";
+    baseUrlPaging: string = "https://localhost:44317/api/ProjectCompare/book?pageSize=5&pageNumber=1";
+    
+    GetBook(pageSize?: number, pageNumber?: number, searchString? : string): Observable<BookModel> {
+      let result: any;
+      //Non Search
+      if(searchString == null){
+        if (pageSize !== undefined && pageNumber !== undefined) {
+          const urlPaging = `${this.baseUrl}/?pageSize=${pageSize}&pageNumber=${pageNumber}`
+          result = this.http.get(urlPaging);
+          return result;
+        } else {
+          result = this.http.get(this.baseUrlPaging);
+          return result
+        }
+      }
+      //Search
+      else if(searchString != null){
+        if (pageSize !== undefined && pageNumber !== undefined) {      
+          const urlPaging =  `https://localhost:44317/api/ProjectCompare/book?pageSize=${pageSize}&pageNumber=${pageNumber}&searchString=${searchString}`
```

Comment:
Use Url variable reference instead. Shouldn't set it statically like this.

Reply:
Can you explain more clearly this issue